bridge-stp.in

```diff
@@ -139,8 +164,59 @@ case $2 in
         # Kill mstpd, since no bridges are currently using it.
         kill $(pidof -c mstpd)
         ;;
-    *)
-        echo "Unknown action: $2" >&2
-        echo "Usage: bridge-stp <bridge> {start|stop}" >&2
-        exit 1
+    restart|restart_config)
+        if [ "$action" = 'restart' ]; then
+            # Kill mstpd.
+            pids="$(pidof -c mstpd)" ; Err=$?
+            if [ $Err -eq 0 ]; then
+                echo 'Stopping mstpd ...'
+                kill $pids
+            fi
+
+            # Start mstpd.
+            echo 'Starting mstpd ...'
+            "$mstpd" $MSTPD_ARGS || exit 3
+        fi
+
+        # Reconfigure bridges.
+        for bridge in $(ls "$net_dir"); do
+            # Ignore interfaces that are not bridges.
+            if [ ! -e "$net_dir/$bridge/bridge/stp_state" ]; then
+                continue
+            fi
+
+            # Ignore bridges that should not use MSTP.
+            if ! is_mstp_bridge "$bridge"; then
+                continue
+            fi
+
+            # Skip bridges that have STP disabled.
+            read State < "$net_dir/$bridge/bridge/stp_state"
+            if [ "$State" = '0' ]; then
+                echo
+                echo "Skipping bridge '$bridge' that has STP disabled." >&2
+                echo "To configure this bridge to use MSTP, run:" >&2
+                echo "brctl stp '$bridge' on" >&2
+                continue
+            fi
+
+            # Skip bridges that are not in user_stp mode.
+            if [ "$State" != '2' ]; then
+                echo
```

Comment:
Actually, I think the echo lines after that should not have >&2 since they are informational messages, not errors.